### PR TITLE
Generators p1

### PR DESCRIPTION
**Pull request by efronbs**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

Issue: https://github.com/yahoo/fili/issues/769

### CODE REVIEW

File: fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/DefaultGranularityGenerator.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+     * @return the generated granularity.
+     */
+    @Override
+    public Granularity bind(
+            DataApiRequestBuilder builder,
+            RequestParameters params,
+            BardConfigResources resources
+    ) {
+        String granularity = params.getGranularity()
+                .orElseThrow(() -> new BadApiRequestException(UNKNOWN_GRANULARITY.logFormat("null")));
```
Comment (michael-mclawhorn) [inline fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/DefaultGranularityGenerator.java]: I'm fairly convinced that this 'null' isn't the right messaging here.

I'm trying to trace the code, is there a code path going through here?

Comment (efronbs) [reply fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/DefaultGranularityGenerator.java]: I will change the message to something else. I was mostly trying to duplicate https://github.com/yahoo/fili/blob/master/fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/ApiRequestImpl.java#L237-L248 , but after looking at it again I think I just need to say granularity was missing from the query.

Comment (michael-mclawhorn) [reply fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/DefaultGranularityGenerator.java]: ultimately, we're going to probably treat missing granularity as meaning all granularity eventually.

Is there any code path that does down this path?

Comment (efronbs) [reply fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/DefaultGranularityGenerator.java]: Currently all data requests must have granularity to even hit the data servlet. Under normal circumstances this code path should never be hit.

**Pull request merged**